module/state_synchronization/execution_data_service.go

```diff
@@ -28,13 +28,17 @@ type BlobTree [][]cid.Cid
 
 // ExecutionDataService handles adding/getting execution data to/from a blobservice
 type ExecutionDataService interface {
+	module.ReadyDoneAware
 	// Add constructs a blob tree for the given ExecutionData and
 	// adds it to the blobservice, and then returns the root CID
 	// and list of all CIDs.
 	Add(ctx context.Context, sd *ExecutionData) (flow.Identifier, BlobTree, error)
 
 	// Get gets the ExecutionData for the given root CID from the blobservice.
 	Get(ctx context.Context, rootID flow.Identifier) (*ExecutionData, error)
+
+	// Check checks if the ExecutionData for the given root CID is valid, and if not, returns the invalid CIDs
+	Check(ctx context.Context, rootID flow.Identifier) ([]cid.Cid, []error)
```

Comment:
Yea, validity is checked before writing it to disk. This was added to support the `execution-data-startup-check` which checks all configured heights to make sure the data exists in the store, and is valid. The intention is to make sure data wasn't corrupted on disk since the files were downloaded, and allow recovering without deleting and starting over.